kombu/transport/consul.py

```diff
@@ -0,0 +1,272 @@
+"""
+Consul Kombu Transport
+
+It uses Consul.io's Key/Value store to transport messages in Queues
+
+It uses python-consul for talking to Consul's HTTP API
+"""
+from __future__ import absolute_import, unicode_literals
+
+import uuid
+import socket
+
+from . import virtual
+from kombu.exceptions import ChannelError
+from kombu.log import get_logger
+from kombu.five import Empty, monotonic
+from kombu.utils.json import loads, dumps
+
+try:
+    import consul
+except ImportError:
+    consul = None
+
+LOGGER = get_logger('kombu.transport.consul')
+
+DEFAULT_PORT = 8500
+DEFAULT_HOST = 'localhost'
+
+
+class Channel(virtual.Channel):
+    """
+    Consul Channel class which talks to the Consul Key/Value store
+    """
+    prefix = 'kombu'
+    index = None
+    timeout = '10s'
+    session_ttl = 30
+    lock_name = '%s' % socket.gethostname()
```

Comment:
moved to cached_property to avoid calling socket.gethostname as a side effect of importing the module